SparkPost/2web2ui
# SA-1274 + SA-1275 Add Timezone and Precision picker to ReportOptions

### What Changed
 - Added timezone picker
 - Forced UTC timezone when precision is set to a precision less or equally precise than 1 day.
 - Added precision selector to the report options.
 - Added selectedPrecision state to Datepicker.
 - Changed logic in Manual Entry Form.
 - Changed many metrics helper functions to include precision as a variable.
 - Added new function in metrics helper.
 - Added new configs for recommended time and precision for metrics rollup

### How To Test
 - Choose a timezone in the picker, ensure requests sent include that timezone
 - Set precision to 1 day (or higher), ensure the timezone is set to UTC, disabled, and has a tooltip explaining why
- Add `x-msys-metrics-rollup` as user option by making a PUT request to '/api/v1/users/{username}/control ' endpoint and with 
```json
{ "options" : { "ui" :{ "use-metrics-rollup": true}	}}
 ```  
 - Add X-Msys-Metrics-Rollup to allowed header in your cors file in openresty
 - Navigate to summary report.
 - Change data ranges and precision around. It should meet AC.
 - Go back to an account without feature. Verify that old functionality has not changed.
